src/app/core/components/language-mismatch/language-mismatch.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { mapPending } from 'src/app/shared/operators/map-pending';
+import { UserSessionService } from 'src/app/shared/services/user-session.service';
+import { LocaleService } from '../../services/localeService';
+
+@Component({
+  selector: 'alg-language-mismatch',
+  templateUrl: './language-mismatch.component.html',
+  styleUrls: [ './language-mismatch.component.scss' ],
+})
+export class LanguageMismatchComponent {
+
+  readonly currentLanguage = this.localeService.currentLang?.tag;
+  readonly languages = this.localeService.languages;
+
+  languageMismatch$ = this.sessionService.userProfile$.pipe(
+    map(profile => {
+      const hasLanguageMismatch = !!this.currentLanguage && profile.defaultLanguage !== this.currentLanguage;
+      return hasLanguageMismatch ? {
+        userDefaultLanguage: profile.defaultLanguage,
+        userDefaultLanguageIsSupported: this.languages.some(({ tag }) => tag === profile.defaultLanguage),
+      } : undefined;
+    }),
+  )
```

Comment:
Code style: 
- `!!this.currentLanguage` is useless in  `!!this.currentLanguage && profile.defaultLanguage !== this.currentLanguage;` as it would not match anyway with "===" if it was undefined. (except if you want to exclude the empty string but I do not see the point)
- so L20 becomes `const hasLanguageMismatch = profile.defaultLanguage !== this.currentLanguage;`
- so no need to define an extra constant
- so there is one line of code in the block 
- so we could just write:
```suggestion
    map(profile => (profile.defaultLanguage !== this.currentLanguage ? {
      userDefaultLanguage: profile.defaultLanguage,
      userDefaultLanguageIsSupported: this.languages.some(({ tag }) => tag === profile.defaultLanguage),
    } : undefined)),
  )
```

Reply:
Fixed